src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelEMWorkspaceMathUtils.java

```diff
@@ -0,0 +1,117 @@
+package org.broadinstitute.hellbender.tools.coveragemodel;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.math3.linear.*;
+import org.apache.commons.math3.util.Precision;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.tools.coveragemodel.nd4jutils.Nd4jApacheAdapterUtils;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.linalg.ops.transforms.Transforms;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Common math utilities for {@link CoverageModelEMWorkspace}
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class CoverageModelEMWorkspaceMathUtils {
+
+    /**
+     * Inverts a square INDArray matrix using Apache LU decomposition
```

Comment:
They were not available when I started the project; they added BLAS level 3 only recently. Luckily, we are not inverting large matrices but rather a large number of small matrices, so the performance is not affected bigly :D I will make an issue to (1) rewrite the Apache matrix operations using Nd4j routines, and (2) use Nd4j workspaces for memory management.